kcidb/oo/__init__.py

```diff
@@ -345,9 +345,9 @@ def repo_branch_checkouts(self):
                 branch_checkouts[branch] = []
             branch_checkouts[branch].append(checkout)
         # Sort branches by number of checkouts
-        for repo in repo_branch_checkouts:
+        for repo, obj in repo_branch_checkouts.items():
             repo_branch_checkouts[repo] = dict(sorted(
-                repo_branch_checkouts[repo].items(),
+                obj.items(),
```

Comment:
Hmm, we don't actually prevent modifying the `repo_branch_checkouts` here. The only difference is that we're accessing its elements via `items()`, and are using an intermediate variable for the values.

Does that really avoid the pylint warning? Is that because we simply outsmarted pylint, or does `items()` actually make a copy of the dictionary keys and values for iteration, and thus avoids the "modify in a loop" problem? Could you check that please?

Thanks!

Reply:
Also, is it really unsafe to change values when iterating over keys?

Reply:
> Does that really avoid the pylint warning? Is that because we simply outsmarted pylint, or does `items()` actually make a copy of the dictionary keys and values for iteration, and thus avoids the "modify in a loop" problem? Could you check that please?
> 
Yes, `.items()` returns an independent copy of the container.  [link(https://stackoverflow.com/a/2315529/13712347). It returns a dictionary view object which is a read-only proxy(an interface to the object in memory), when the dictionary changes, the view also reflect these changes.

Reply:
> Also, is it really unsafe to change values when iterating over keys?

No
https://stackoverflow.com/a/6777632/13712347

`Pylint` probably assumes that any changes to the original container might mean modification, and hence the need to iterate over a copy, then change the original container.

Reply:
Looking at it and reading some more, I think it's just fine to leave as is, and ask pylint to not worry about this :grin:

We're not change the dictionary keys, only values, so that's OK, and the original code is simpler.

Thank you!

Reply:
Okay.